### ISSUE TITLE
`new` function confuses object construction

### ISSUE THREAD

**Issue by nenadalm** [clj-kondo/clj-kondo#1731]
**version**

```shell
$ clj-kondo --version
clj-kondo v2022.05.31
```

**platform**

native

**problem**

If `new` function is defined, it confuses clj-kondo.

**repro**

`src/app/repro.cljs`:
```clojure
(ns app.repro)

(defn new [])

(js/Date. 2022 1 1 1 1)
```

```shell
$ clj-kondo --lint src/
src/app/repro.cljs:5:1: error: app.repro/new is called with 6 args but expects 0
linting took 8ms, errors: 1, warnings: 0
```

**expected behavior**

Since the code seems to be working, I don't think defining `new` function is problematic and there should be no error about wrong arity of the `js/Date` constructor.

**Comment by borkdude** [comment]
I agree that this is a bug and we should fix it. But what I don't understand is why do you have a `new` function?

``` clojure
(ns app.repro)

(defn new [] :foo)

(new js/Date 2022 1 1 1 1)

(new) ;; this throws
```

**Comment by nenadalm** [comment]
You're right that calling the function from the same namespace doesn't work. If you call it from different namespace though - it works.

> But what I don't understand is why do you have a new function?

It's in project that we took over and it's there since 4 years ago. I'll probably rename, but I thought that the message is a bit confusing, so I created the issue.

**Comment by sumbach** [comment]
See also #686 / #1709 -- I think the change to `expand-dot-constructor` in that PR allows for ignoring `:invalid-arity` when using the dot syntax:

```
(ns app.repro)

(defn new [] :foo)

#_{:clj-kondo/ignore [:invalid-arity]}
(new js/Date 2022 1 1 1 1) ;; no linter error with clj-kondo v2022.04.25 or v2022.06.22

#_{:clj-kondo/ignore [:invalid-arity]}
(js/Date. 2022 1 1 1 1) ;; invalid-arity error with clj-kondo v2022.04.25; no linter error with v2022.06.22
```

**Comment by sumbach** [comment]
Confirmed with clj-kondo v2022.08.03 -- we no longer see invalid-arity errors regarding `app.repro/new` when using `(new SomeClass ,,,)` constructor or the `(SomeClass. ,,,)` syntax.

I've also taken down a TODO to avoid using the name `new` for vars or bindings. Even though this isn't exactly shadowing (in the same way `(defn format [,,,] ,,,)` shadows `clojure.core/format` or `(let [keys [:name :address]] ,,,)` shadows `clojure.core/keys`), this is potentially confusing/ambiguous to someone trying to _read_ the code.

Thanks @borkdude 😄 👍
